### DISCUSSION TITLE
Can -scene increment by 2 instead of 1?

### DISCUSSION THREAD

**Discussion by 00SS** [ImageMagick/ImageMagick#2432] [Help] [answered]
I have a hundred scanned pages of a book. 

Image size: 3000x2500 pixels
Each image has two pages - left and right.
The left pages are odd numbered, the right pages are even numbered.
So the first scan has _page 1_ on the left, _page 2_ on the right.
The second scan, _page 3_ on the left, _page 4_ on the right.
etc.

I want to split the pages, and the new images to have an image number corresponding to the page number.

When I split the images, the results go into a subdirectory /split

Presently I use these 2 commands:

    convert.exe scan_*.tif -gravity West -crop 50%x100%+0+0 -repage 0x0 -deskew 40% -gravity center -extent 1500x2500 -quality 9 -scene 1 split\page_%04d-left.png

The output in subdirectory **\split** is: `page_0001-left.png`  `page_0002-left.png`  `page_0003-left.png` etc.

    convert.exe scan_*.tif -gravity East -crop 50%x100%+0+0 -repage 0x0 -deskew 40% -gravity center -extent 1500x2500 -quality 9 -scene 1 split\page_%04d-right.png

The output in subdirectory **\split** is: `page_0001-right.png`  `page_0002-right.png`  `page_0003-right.png` etc.

What I would like to do is have the -scene option step up the count by 2 instead of 1 - Example (imaginary code): 
`-scene 1x2 page_%04d.png` would give an output: `page_0001.png`  `page_0003.png`  `page_0005.png` etc.
`-scene 2x2 page_%04d.png` would give an output: `page_0002.png`  `page_0004.png`  `page_0006.png` etc.

Is there a way to do this?

**Comment by fmw42** [comment]
I do not know if this will work, but in IM 7 see -define frames:step=2. See https://imagemagick.org/script/defines.php

**Comment by GeeMack** [comment, marked as answer]
You can get a similar result using ImageMagick by setting the output filenames using an FX expression. It uses a workaround for formatting the number of digits since IM has very limited ability to do that. Here's an example command...

    convert logo: -duplicate 9 -set filename:0 %[fx:t*2+1000] page-%[filename:0].png

The output names are "page-1000.png", "page-1002.png", "page-1004.png", etc.

To make odd numbers, use "1001" in that FX expression...

    convert logo: -duplicate 9 -set filename:0 %[fx:t*2+1001] page-%[filename:0].png

Obviously the filename numbering is 1001, 1003, 1005, ..., instead of 0001, 0003, 0005. It might be the closest you can get without going to a "for" loop in your OS shell.

**Comment by 00SS** [comment]
@GeeMack

For the left hand page I used : `-set filename:0 %[fx:t*2+1001] split\page_%[filename:0].png`
The output in subdirectory **\split** was:
`page_1001.png`  `page_1003.png`  `page_1005.png`  etc.

For the right hand page I used: `-set filename:0 %[fx:t*2+1002] split\page_%[filename:0].png`
This produced:
`page_1002.png`  `page_1004.png`  `page_1006.png`  etc.

Each output image's file name's numerical ending corresponding with the scan's page number.

This is PERFECTLY what I wanted ! ! !

Thank you so much.


_Note: It could be useful if the `-scene` option had a way of setting an increment step value_
